lib/webhdfs/client_v1.rb

```diff
@@ -59,32 +61,62 @@ def initialize(host='localhost', port=50070, username=nil, doas=nil, proxy_addre
 
       @kerberos = false
       @kerberos_keytab = nil
+      @renew_kerberos_delegation_token_time_hour = renew_kerberos_delegation_token_time_hour
+      @kerberos_delegation_token = nil
+      @kerberos_token_updated_at = Time.now
       @http_headers = http_headers
     end
 
+    def should_kerberos_token_updated?
+      @kerberos_token_updated_at + (@renew_kerberos_delegation_token_time_hour * 60 * 60) <= Time.now
+    end
+
+
+    def get_cached_kerberos_delegation_token
+      return @kerberos_delegation_token if !@kerberos_delegation_token.nil? && !should_kerberos_token_updated?
```

Comment:
In Ruby, the only `nil` and `false` are falsy values. Otherwise, true.
So this statement should be `if @kerberos_delegation_token && !should_kerberos_token_updated?`.